### PR TITLE
(fix) updated the procedures app UI by grouping orders by patient

### PR DESCRIPTION
**Pull request by FelixKiprotich350**
## Requirements
- [ ] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.

## Summary
fixed the grouping of orders by patient and displaying the name of the patient alongside his/her total orders pending

## Screenshots
**screen shot**
Screenshot from 2024-07-29 15-48-50
**screen record**
[procedures-grouping.webm](https://github.com/user-attachments/assets/478dab69-7e31-46b5-8265-5593c2bd9bf4)

### CODE REVIEW

**Comment by FelixKiprotich350** [comment]
@makombe  , @donaldkibet and @ojwanganto have a look

**Comment by ojwanganto** [comment]
Thanks, @FelixKiprotich350. cc @makombe @donaldkibet

File: src/common/commonProps.resource.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  groupedOrders: GroupedOrders;
+  showStatus: boolean;
+  showActions: boolean;
+  showOrderType: boolean;
+  actions: Array<OrderAction>;
+}
+
+export interface OrderAction {
+  actionName: string;
+  displayPosition: 0 | number;
```
Comment (makombe) [inline src/common/commonProps.resource.ts:44]: we can name this file appropriately, seems to be types and not resource

File: src/common/listOrderDetails.scss [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    color: #525252;
+    font-weight: bold;
+    width: layout.$spacing-05;
+    height: layout.$spacing-05;
+  }
+  
+  .actionBtns {
+    float: right;
+    margin-left: 1rem;
+    margin-top: 2rem;
```
Comment (makombe) [inline src/common/listOrderDetails.scss:54]: Preferably use carbon spacing

File: src/common/groupedOrdersTable.component.tsx [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+            >
+              <TableToolbarContent>
+                {props.showStatusFilter && (
+                  <Layer style={{ margin: "5px" }}>
+                    <Dropdown
+                      id="orderStatus"
+                      initialSelectedItem={"All"}
+                      label=""
+                      titleText={
+                        t("filterOrdersByStatus", "Filter Orders by status") +
```
Comment (makombe) [inline src/common/groupedOrdersTable.component.tsx:150]:
```suggestion
                        t("filterOrdersByStatus", "Filter orders by status") +
```

File: src/common/groupedOrdersTable.component.tsx [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+                    </React.Fragment>
+                  ))
+                ) : (
+                  <TableRow>
+                    <TableCell
+                      colSpan={headers.length + 1}
+                      style={{ textAlign: "center" }}
+                      className={styles.noOrdersDiv}
+                    >
+                      {t("noOrderAvailable", "No Orders Availalble")}
```
Comment (makombe) [inline src/common/groupedOrdersTable.component.tsx:229]:
```suggestion
                      {t("noOrderAvailable", "No orders available")}
```

**Review by makombe — COMMENTED** [review]
Thanks @FelixKiprotich350

File: src/common/groupedOrdersTable.component.tsx (hunk 4 above)
Comment (makombe) [reply src/common/groupedOrdersTable.component.tsx:229]: Can you also re-attach the video not able to view the attached.

File: src/common/commonProps.resource.ts (hunk 1 above)
Comment (donaldkibet) [reply src/common/commonProps.resource.ts:44]: types and interfaces should be centralized on `types/index.ts` file

File: src/common/groupedOrdersTable.component.tsx [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+} from "@carbon/react";
+import ListOrderDetails from "./listOrderDetails.component";
+import Overlay from "../components/overlay/overlay.component";
+
+//  render Grouped by patient Orders in procedures app
+const GroupedOrdersTable: React.FC<GroupedOrdersTableProps> = (props) => {
+  const workListEntries = props.orders;
+  const { t } = useTranslation();
+  const [currentPageSize, setCurrentPageSize] = useState<number>(10);
+  const [searchString, setSearchString] = useState<string>("");
```
Comment (donaldkibet) [inline src/common/groupedOrdersTable.component.tsx:36]: We should removed unused imports

File: src/common/groupedOrdersTable.component.tsx [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+    "All",
+    "RECEIVED",
+    "IN_PROGRESS",
+    "COMPLETED",
+    "EXCEPTION",
+    "ON_HOLD",
+    "DECLINED",
+  ];
+
+  const [filter, setFilter] = useState<
```
Comment (donaldkibet) [inline src/common/groupedOrdersTable.component.tsx:49]: Maybe having this as a type e.g 

```tsx
type FilterType = 
  | "All"
  | "EXCEPTION"
  | "RECEIVED"
  | "COMPLETED"
  | "IN_PROGRESS"
  | "ON_HOLD"
  | "DECLINED";

const [filter, setFilter] = useState<FilterType>("All");
```

Comment (donaldkibet) [reply src/common/groupedOrdersTable.component.tsx:49]: we can extract and store in the file storing types

File: src/common/groupedOrdersTable.component.tsx [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+    groupedOrdersByPatient,
+    searchString
+  );
+  const {
+    goTo,
+    results: paginatedResults,
+    currentPage,
+  } = usePagination(searchResults, currentPageSize);
+
+  const pageSizes = [10, 20, 30, 40, 50];
```
Comment (donaldkibet) [inline src/common/groupedOrdersTable.component.tsx:105]: use `usePaginationInfo` hook to generate this pageSizes its imported from `@openmrs/esm-framework`

File: src/common/groupedOrdersTable.component.tsx [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+
+  const rowsdata = useMemo(() => {
```
Comment (donaldkibet) [inline src/common/groupedOrdersTable.component.tsx:107]:
```suggestion
  const rowsData = useMemo(() => {
```

File: src/common/groupedOrdersTable.component.tsx [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+    return paginatedResults.map((patient) => ({
+      id: patient.patientId,
+      patientname: patient.orders[0].patient?.display?.split("-")[1],
```
Comment (donaldkibet) [inline src/common/groupedOrdersTable.component.tsx:110]:
```suggestion
      patientName: patient.orders[0].patient?.display?.split("-")[1],
```
please use camelCase

File: src/common/groupedOrdersTable.component.tsx [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+      orders: patient.orders,
+      totalorders: patient.orders?.length,
+    }));
+  }, [paginatedResults]);
+
+  const tableColumns = [
+    { id: 0, header: t("patient", "Patient"), key: "patientname" },
```
Comment (donaldkibet) [inline src/common/groupedOrdersTable.component.tsx:117]:
```suggestion
    { id: 0, header: t("patient", "Patient"), key: "patientName" },
```

File: src/common/groupedOrdersTable.component.tsx [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+                        onChange={(event) => {
+                          setActivatedOnOrAfterDate(event.target.value);
+                        }}
+                        type="date"
+                        value={activatedOnOrAfterDate}
+                      />
+                    </DatePicker>
+                  </Layer>
+                )}
+                <Layer style={{ margin: "5px" }}>
```
Comment (donaldkibet) [inline src/common/groupedOrdersTable.component.tsx:175]: please extract styles to styles sheet

File: src/common/groupedOrdersTable.scss [hunk 12]
```diff
@@ -0,0 +1,46 @@
+@use '@carbon/layout';
+@use '@carbon/colors';
+@use '@openmrs/esm-styleguide/src/vars' as *;
+@use '@carbon/styles/scss/spacing';
```
Comment (donaldkibet) [inline src/common/groupedOrdersTable.scss:4]:
```suggestion
@use '@carbon/layout';
```

**Review by donaldkibet — CHANGES_REQUESTED** [review]

**Comment by FelixKiprotich350** [comment]
@donaldkibet , @ojwanganto  and @makombe .... verify if l left out something on the requested changes

**Review by donaldkibet — APPROVED** [review]

**Pull request merged**